src/profile/helpers/__tests__/updateMutationVariables.test.ts

```diff
@@ -106,112 +108,251 @@ const newAddress = {
-  phones: [primaryPhone, newPhone, updatedSecondaryPhone],
-  primaryPhone,
-  addresses: [primaryAddress, newAddress, updatedSecondaryAddress],
-  primaryAddress,
-};
+type FormParts = (keyof FormValues)[];
 
-test('add arrays are formed correctly with new data', () => {
-  const variables: UpdateMyProfileVariables = updateMutationVariables(
-    formValues,
-    myProfile
-  );
+const multiNodeFormParts: FormParts = [
+  'primaryAddress',
+  'primaryEmail',
+  'primaryPhone',
+  'phones',
+  'emails',
+  'addresses',
+];
 
-  expect(variables.input.profile.addAddresses).toEqual([
-    getAddressAsComparisonObject(newAddress),
-  ]);
+const nameFormParts: FormParts = ['firstName', 'nickname', 'lastName'];
 
-  expect(variables.input.profile.addEmails).toEqual([
-    getEmailAsComparisonObject(newEmail),
+const getFormValues = (parts: FormParts): Partial<FormValues> => {
+  const myProfileNamesAndLanguage = _.pick(myProfile.myProfile, [
+    ...nameFormParts,
+    'language',
   ]);
+  const formValues = {
+    ...myProfileNamesAndLanguage,
+    primaryEmail,
+    emails: [primaryEmail, newEmail, updatedSecondaryEmail],
+    phones: [primaryPhone, newPhone, updatedSecondaryPhone],
+    primaryPhone,
+    addresses: [primaryAddress, newAddress, updatedSecondaryAddress],
+    primaryAddress,
+  };
+  return _.pick(formValues, parts) as Partial<FormValues>;
+};
 
-  expect(variables.input.profile.addPhones).toEqual([
-    getPhoneAsComparisonObject(newPhone),
-  ]);
-});
+describe('Update variables for basic data (names) are formed correctly', () => {
+  let formValues: Partial<FormValues>;
 
-test('add arrays are empty when using existing data', () => {
-  const variables: UpdateMyProfileVariables = updateMutationVariables(
-    {
-      ...formValues,
-      addresses: [primaryAddress, updatedSecondaryAddress],
-      emails: [primaryEmail, updatedSecondaryEmail],
-      phones: [primaryPhone, updatedSecondaryPhone],
-    },
-    myProfile
-  );
-
-  expect(variables.input.profile.addPhones).toEqual([]);
-  expect(variables.input.profile.addEmails).toEqual([]);
-  expect(variables.input.profile.addAddresses).toEqual([]);
-});
+  beforeEach(() => {
+    formValues = getFormValues(nameFormParts);
+  });
 
-test('update arrays are formed correctly with updated data objects', () => {
-  const variables: UpdateMyProfileVariables = updateMutationVariables(
-    formValues,
-    myProfile
-  );
-  expect(variables.input.profile.updateAddresses).toEqual([
-    getAddressAsComparisonObject(updatedSecondaryAddress),
-  ]);
-  expect(variables.input.profile.updateEmails).toEqual([
-    getEmailAsComparisonObject(updatedSecondaryEmail),
-  ]);
-  expect(variables.input.profile.updatePhones).toEqual([
-    getPhoneAsComparisonObject(updatedSecondaryPhone),
-  ]);
+  test('Variables in update data match formValues even if not changed', () => {
+    const variables: UpdateMyProfileVariables = updateMutationVariables(
+      formValues,
+      myProfile
+    );
```

Comment:
This variables initialisation seems to be done in multiple tests (in 7 total). Not sure if it could be initialised in beforeEach, and then in tests that do not wish to use it, they would reinitialise it differently. Though sometimes repetition actually improves readability it feels, so not sure 🤔

Reply:
Yeah, there are 7 of those code blocks and in between 5 that does different call. That's why the call is not in beforeEach. Those 7 cannot be logically grouped under one describe(). I thought about using a helper function, but then the readability came to mind. Would it be harder to follow the test if some tests had different initialisation function than others?

Instead of 

```
const variables: UpdateMyProfileVariables = updateMutationVariables(
      formValues,
      myProfile
    );
```
it was something like

`createCommonUpdateVariables();`

Or create one helper function for all tests - with optional formValues:

`createUpdateVariables(formValues?:FormValues);`

Better or worse? :)

Reply:
One option could be to make 1 more master describe on top of those all, and inside that there would be beforeEach. Not sure if it would make things more complicated though. Repetition of those initialisations like it is now might be best solution for readability

Reply:
Left as it is.